### ISSUE TITLE
[FLI-432] Flipt cannot work with PGBouncer in transaction pooling mode

### ISSUE THREAD

**Issue by sashayakovtseva** [flipt-io/flipt#1750]
### Describe the Bug

Flipt doesn't work with Postgres behind a PGBouncer in transaction pooling mode. Using `binary_parameters=yes` does not help.

### Version Info

v1.22.1

### To Reproduce

1) setup postgres
2) setup pgbouncer in transaction pooling mode
3) connect Flipt to pgbouncer. make auth required. enable tokens.
4) try to login

### Expected Behavior

Flipt successfully communicates with postgres. 
You are able to login with a bootstrap token. 
You are able to use PGBouncer in transaction pooling mode.

### Additional Context

As far as I understand, the issue is in StmtCacher https://github.com/flipt-io/flipt/blob/main/internal/storage/sql/db.go#L43 which uses Prepare underneath. 

Flipt logs: 
```
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"4\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"finished unary call with code Internal","server":"grpc","grpc.start_time":"2023-06-14T11:38:48Z","system":"grpc","span.kind":"server","grpc.service":"flipt.Flipt","grpc.method":"GetNamespace","peer.address":"127.0.0.1:48412","error":"rpc error: code = Internal desc = pq: prepared statement \"10\" already exists","grpc.code":"Internal","grpc.time_ms":1.416}
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:38:48Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:38:49Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:50:19Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
{"level":"ERROR","ts":"2023-06-14T11:55:25Z","msg":"unauthenticated","server":"grpc","reason":"error retrieving authentication for client token","error":"getting authentication by token: reading authentication: pq: prepared statement \"3\" does not exist"}
```

<sub>[FLI-432](https://linear.app/flipt/issue/FLI-432/flipt-cannot-work-with-pgbouncer-in-transaction-pooling-mode)</sub>

**Comment by GeorgeMac** [comment]
Thanks for the issue @sashayakovtseva 🙏 

This appears to be a known issue with PGBouncer in transaction pooling mode and prepared statements:
https://www.pgbouncer.org/faq.html#how-to-use-prepared-statements-with-transaction-pooling

It looks like we need a way to configure disabling prepared statements in Flipt to support this.
Or we might need a subtle change and to only support session pooling, not sure which is best.

**Comment by sashayakovtseva** [comment]
I personally vote for configuration option + documentation on the current behaviour :)

**Comment by markphelps** [comment]
Actually I think:

```
database:
  prepared_statements_enabled: true | false (default: true)
```

Might be nicer and match existing config structure

**Comment by markphelps** [comment]
@sashayakovtseva we just added a new option `db.prepared_statements_enabled` which you can set to `false` to get PGBouncer working.

This is available in the latest release [v1.23.1](https://github.com/flipt-io/flipt/releases/tag/v1.23.1)

Im working on updating our docs now to reflect this. Please let us know if this fixes your issue and thanks for reporting and using Flipt!!

**Comment by sashayakovtseva** [comment]
thanks for such a quick reaction to this issue!
